docs/references/ic-interface-spec.md

```diff
@@ -2202,6 +2223,67 @@ These system calls return costs in Cycles, represented by 128 bits, which will b
     - `1`: Invalid curve or algorithm. Memory at `dst` is left unchanged.
     - `2`: Invalid key name for the given combination of signing scheme and (valid) curve/algorithm. Memory at `dst` is left unchanged.
 
+### Environment Variables
+
+The following system calls provide access to the canister's environment variables:
+
+-   `ic0.env_var_count : () -> I`; `I ∈ {i32, i64}`
+
+    Returns the number of environment variables set for this canister.
+
+-   `ic0.env_var_name_size : (index: I) -> I`; `I ∈ {i32, i64}`
+
+    Gets the size in bytes of the name of the environment variable at the given index.
+
+    This system call traps if:
+      - If the index is out of bounds (>= than value provided by `ic0.env_var_count`)
+
+-   `ic0.env_var_name_copy : (index: I, dst: I, offset: I, size: I) -> ()`; `I ∈ {i32, i64}`
```

Comment:
it would be helpful to elaborate on the arguments here. i would have assumed the index and size are enough to retrieve the  copy. what are `dst` and `offset` needed for?